### ISSUE TITLE
KeycloackRestTemplate with service discovery capabilities

### ISSUE THREAD

**Issue by amarchese96** [ihuaylupo/manning-smia#19]
Is it possible to add service discovery capabalities to KeycloackRestTemplate class? I tried to annotate the KeycloackRestTemplate bean definition with @LoadBalanced annotation, but it seems that the KeycloackRestTemplate doesn’t resolve service name using Eureka Discovery Server.

**Comment by krishnaramachandran1973** [comment]
Yes, same issue here too. java.net.UnknownHostException: No such host is known (gateway-server). Using Keycloak versin 15.0.2

**Comment by rkrishnnakumar** [comment]
Removing @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE) made it work. Thanks.
        @Bean
	@LoadBalanced
	public KeycloakRestTemplate keycloakRestTemplate() {
		return new KeycloakRestTemplate(keycloakClientRequestFactory);
	}
